### PR TITLE
More files with CanGc fixes

### PR DESCRIPTION
**Pull request by chickenleaf**
**File Changes**: 
Replaced `CanGc::note()` calls in these files + made required entries in `Bindings.conf`

1. `components/script/dom/urlsearchparams.rs`
2. `components/script/dom/submitevent.rs`
3. `components/script/dom/webglcontextevent.rs`
4. `components/script/dom/rtcdatachannelevent.rs`
5. `components/script/dom/rtcicecandidate.rs`
6. `components/script/dom/xrrigidtransform.rs`
7. `components/script/dom/messageevent.rs`

**Files that couldn't be changed**:
1. `components/script/dom/htmlmediaelement.rs` - all inside `VirtualMethods`
2. `components/script/dom/globalscope.rs` - inside `!task`
3. `components/script/dom/htmlsourceelement.rs`- all inside `VirtualMethods` or `FetchResponseListener`

**Note**:
I'm not so sure about this PR please check the fixes made through the propagation of the CanGc fix from `messageevent.rs` line 161 `CanGc::note()`, while fixing that there were changes that led to adding few `CanGc::note()` inside `!task` and in calls within `FetchResponseListener` (NO FUNCTION PARAMETERS WERE ADDED/MODIFIED)

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are part of #33683

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because they do not modify functionality

### CODE REVIEW

**Comment by chickenleaf** [comment]
@jdm please review the changes made during the fixing of `messageevent.rs` line 161 `CanGc::note()`. If you'd like me to send in a separate PR just containing those changes I could do that, if that would make the review easier and thorough. Please let me know!

File: components/script/dom/dedicatedworkerglobalscope.rs [hunk 1]
```diff
@@ -608,13 +609,14 @@ impl DedicatedWorkerGlobalScope {
...
         transfer: CustomAutoRooterGuard<Vec<*mut JSObject>>,
+        can_gc: CanGc,
     ) -> ErrorResult {
         let data = structuredclone::write(cx, message, Some(transfer))?;
         let worker = self.worker.borrow().as_ref().unwrap().clone();
         let global_scope = self.upcast::<GlobalScope>();
         let pipeline_id = global_scope.pipeline_id();
         let task = Box::new(task!(post_worker_message: move || {
-            Worker::handle_message(worker, data);
+            Worker::handle_message(worker, data, can_gc);
```
Comment (jdm) [inline components/script/dom/dedicatedworkerglobalscope.rs:619]: This is within a task! macro, so we should not propagate it.

**Review by jdm — CHANGES_REQUESTED** [review]
No concerns about the message event changes!

File: components/script/dom/dedicatedworkerglobalscope.rs (hunk 1 above)
Comment (chickenleaf) [reply components/script/dom/dedicatedworkerglobalscope.rs:619]: i shall fix that

**Review by jdm — APPROVED** [review]

**Pull request merged**